# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/Worklet.h
# Commit: 4bed3a0e2332
# Full Hash: 4bed3a0e2332f24ffe38f08e234c8898c42249ee
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2016-11-07 02:21:38
# Regressor Bug: 1290021
# File Overlap Count: 1
# Description:
#   Bug 1290021 - Implement a prototype version of Houdini "Worklets Level 1" spec - part 1 - WebIDL interface, r=smaug
# ==============================================================================

diff -r d84108f190ee -r 4bed3a0e2332 dom/worklet/Worklet.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/worklet/Worklet.h	Sun Nov 06 09:53:34 2016 +0100
@@ -0,0 +1,54 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_Worklet_h
+#define mozilla_dom_Worklet_h
+
+#include "mozilla/Attributes.h"
+#include "mozilla/ErrorResult.h"
+#include "nsWrapperCache.h"
+#include "nsCOMPtr.h"
+
+class nsIGlobalObject;
+
+namespace mozilla {
+namespace dom {
+
+class Promise;
+
+class Worklet final : public nsISupports
+                    , public nsWrapperCache
+{
+public:
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
+  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(Worklet)
+
+  explicit Worklet(nsIGlobalObject* aGlobal)
+    : mGlobal(aGlobal)
+  {}
+
+  nsIGlobalObject* GetParentObject() const
+  {
+    return mGlobal;
+  }
+
+  virtual JSObject*
+  WrapObject(JSContext* aCx, JS::Handle<JSObject*> aGivenProto) override;
+
+  already_AddRefed<Promise>
+  Import(const nsAString& aModuleURL, ErrorResult& aRv);
+
+private:
+  ~Worklet()
+  {}
+
+  nsCOMPtr<nsIGlobalObject> mGlobal;
+};
+
+} // namespace dom
+} // namespace mozilla
+
+#endif // mozilla_dom_Worklet_h